sqlmesh/core/model/kind.py

```diff
@@ -407,7 +362,42 @@ class IncrementalByTimeRangeKind(_IncrementalBy):
     time_column: TimeColumn
     auto_restatement_intervals: t.Optional[SQLGlotPositiveInt] = None
 
-    _time_column_validator = TimeColumn.validator()
+    @field_validator("time_column", mode="before")
+    @classmethod
+    def _time_column_validator(cls, v: t.Any, values: t.Any) -> TimeColumn:
```

Comment:
I think it was intentionally defined as part of the `TimeColumn` class to keep it closer to the actual model definition. Can we keep it there?

Reply:
Why is it using `values: t.Any` and not `ValidationInfo`?

Reply:
Updated.